native-sql-engine/core/src/main/scala/com/intel/oap/execution/ColumnarShuffledHashJoinExec.scala

```diff
@@ -356,31 +356,47 @@ case class ColumnarShuffledHashJoinExec(
 
       // now we can return this wholestagecodegen iter
       val res = new Iterator[ColumnarBatch] {
-        override def hasNext: Boolean = iter.hasNext
+        override def hasNext: Boolean = {
+          nativeIterator.hasNext || iter.hasNext
+        }
 
         override def next(): ColumnarBatch = {
-          val cb = iter.next()
-          val beforeEval = System.nanoTime()
-          val input_rb =
-            ConverterUtils.createArrowRecordBatch(cb)
-          if (input_rb.getLength == 0) {
-            ConverterUtils.releaseArrowRecordBatch(input_rb)
+          if (nativeIterator.hasNext) {
```

Comment:
should also release the input batch in this case? 
`ConverterUtils.releaseArrowRecordBatch(input_rb)`